### PR TITLE
Add iframes to the pointer event attribute tests

### LINKED ISSUE #3462: Add pointerevent attribute test inside an iframe
There is no test in pointer event test set to test the attributes of them inside an iframe.
Particularly, since issues like
https://bugs.chromium.org/p/chromium/issues/detail?id=605676
happened in Chrome, it might be worth adding them.

### PR DESCRIPTION
**Pull request by NavidZ**
This helps us test the attributes of the pointerevents
dispatched inside an iframe.

closes #3462

### CODE REVIEW

File: pointerevents/pointerevent_support.js [hunk 1]
```diff
@@ -12,10 +12,12 @@ var All_Pointer_Events = [
...
 // TA: 1.1, 1.2, 1.6, 1.7, 1.8, 1.9, 1.10, 1.11, 1.12, 1.13
-function check_PointerEvent(event) {
-    var pointerTestName = event.pointerType + ' ' + event.type;
+function check_PointerEvent(event, testNamePrefix, pointerEventInterface) {
+    if (testNamePrefix === undefined)
+        testNamePrefix = "";
+    var pointerTestName = testNamePrefix + ' ' + event.pointerType + ' ' + event.type;
     test(function () {
-        assert_true(event instanceof PointerEvent, "event is a PointerEvent event");
+        assert_true(event instanceof (pointerEventInterface || PointerEvent), "event is a PointerEvent event");
```
Comment (mustaqahmed) [inline pointerevents/pointerevent_support.js]: Any chance the |pointerEventInterface| check is stale?

Comment (NavidZ) [inline pointerevents/pointerevent_support.js]: |pointerEventInterface| is either null (when checking the event that was fired in the outer frame) which "(pointerEventInterface || PointerEvent)" will be "PointerEvent" or it is the pointer event interface inside the inner frame which then the result of "(pointerEventInterface || PointerEvent)" will be pointerEventInterface.

File: pointerevents/pointerevent_attributes_mouse-manual.html [hunk 2]
```diff
@@ -88,9 +107,10 @@
         <h1>Pointer Events pointerdown tests</h1>
         <h4>
             Test Description: This test checks the properties of mouse pointer events. Move your mouse over the black square and click on it. Then move it off the black square.
+            Test passes if the proper behavior of the events is observed.
```
Comment (mustaqahmed) [inline pointerevents/pointerevent_attributes_mouse-manual.html]: Please expand into 4 steps: click, move, click, move. Perhaps mark the iframe through border/background color and say something like "...click it again when in red..."

File: pointerevents/pointerevent_styles.css [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
 }
 
 #square1 {
-background: black;
 top: 150px;
 left: 100px;
 }
 
+#square2 {
+top: 50px;
```
Comment (mustaqahmed) [inline pointerevents/pointerevent_styles.css]: Perhaps make square2 non-black (red?) to make the instruction easier.

File: pointerevents/pointerevent_attributes_mouse-manual.html [hunk 4]
```diff
@@ -17,68 +17,87 @@
...
+                    }, pointerTestName + "'s ClientX and ClientY attributes are correct.");
+                }
+
+                // Test isPrimary
+                test(function () {
+                    assert_equals(event.isPrimary, true, "isPrimary should be true");
+                }, pointerTestName + ".isPrimary attribute is correct.");
+
+                check_PointerEvent(event, testNamePrefix, pointerEventInterface);
+                detected_eventTypes[event.type] = true;
```
Comment (mustaqahmed) [inline pointerevents/pointerevent_attributes_mouse-manual.html]: Please move back <code>detected_eventTypes[event.type] = true<code> right after the <code>if</code> to the beginning of this method: makes the reentry-prevention easy to follow.

File: pointerevents/pointerevent_attributes_mouse-manual.html [hunk 5]
```diff
@@ -17,68 +17,87 @@
...
+                        if (Object.keys(detected_eventTypes).length == eventList.length) {
+                            square1.style.visibility = 'hidden';
+                            detected_eventTypes = {};
+                            square2.style.visibility = 'visible';
                         }
-
-                        // Test clientX and clientY
-                        if (event.type != 'pointerout' && event.type != 'pointerleave' ) {
-                            test(function () {
-                                assert_true(event.clientX >= rectSquare1.left && event.clientX < rectSquare1.right, "ClientX should be in the boundaries of the black box");
```
Comment (mustaqahmed) [inline pointerevents/pointerevent_attributes_mouse-manual.html]: Please combine clientX/Y tests, as in the <code>else</code> block below.

File: pointerevents/pointerevent_attributes_mouse-manual.html [hunk 6]
```diff
@@ -17,68 +17,87 @@
...
+                    }, pointerTestName + "'s button is -1 when mouse buttons are released.");
+                    test(function() {
+                        assert_true(event.buttons == 0, "If mouse buttons are released buttons attribute is 0")
+                    }, pointerTestName + "'s buttons is 0 when mouse buttons are released.");
+                }
+
+                // Test clientX and clientY
+                if (event.type != 'pointerout' && event.type != 'pointerleave' ) {
+                    test(function () {
+                        assert_true(event.clientX >= targetBoundingClientRect.left && event.clientX < targetBoundingClientRect.right, "ClientX should be in the boundaries of the black box");
```
Comment (mustaqahmed) [inline pointerevents/pointerevent_attributes_mouse-manual.html]: Please combine clientX/Y tests, as in the else block below.

File: pointerevents/pointerevent_attributes_mouse-manual.html [hunk 7]
```diff
@@ -17,68 +17,87 @@
...
+                    }, pointerTestName + "'s buttons attribute is 1 when left mouse button is pressed.");
+                } else if (event.type == 'pointerup') {
+                    test(function() {
+                        assert_true(event.button == 0, "If left mouse button is just released button attribute is 0")
+                    }, pointerTestName + "'s button attribute is 0 when left mouse button is just released.");
+                    test(function() {
+                        assert_true(event.buttons == 0, "If left mouse button is just released buttons attribute is 0")
+                    }, pointerTestName + "'s buttons attribute is 0 when left mouse button is just released.");
+                } else {
+                    test(function() {
```
Comment (mustaqahmed) [inline pointerevents/pointerevent_attributes_mouse-manual.html]: The texts in the following two tests are misleading: these are not related to mouse-button release activity at all. Instead, they should simply say "pointerout's button should be x" etc.

While doing this, please consider combining all button/buttons into a single test per event type. We have 515 asserts in total, perhaps inherited ones should be combined to shorten the long list of PASSes.

**Comment by mustaqahmed** [comment]
Overall, your patch looks good. However, the existing asserts needs some attention IMO, so asked you to clean them up in the same go.

File: pointerevents/pointerevent_attributes_mouse-manual.html (hunk 7 above)
Comment (NavidZ) [inline pointerevents/pointerevent_attributes_mouse-manual.html]: I updated the test descriptions. I hope this one looks better.
Regarding merging button and buttons tests I believe it is better to keep them separate. I see no problem with having a lot of small tests. As long as user doesn't need to do separate stuff for them. Particularly button and buttons seem different enough to have different tests for them.

File: pointerevents/pointerevent_attributes_mouse-manual.html (hunk 6 above)
Comment (NavidZ) [inline pointerevents/pointerevent_attributes_mouse-manual.html]: I combined the clientX/Y tests into one as they are related and there was only one test anyway for the pointerout/leave case.

File: pointerevents/pointerevent_support.js (hunk 1 above)
Comment (RByers) [inline pointerevents/pointerevent_support.js]: Rather than pass this in, could you just check `event instanceof event.target.ownerDocument.abstractView.PointerEvent`?  Then that would validate that the interface belongs to the same window as the target - a nice extra property to verify explicitly.

**Comment by RByers** [comment]
Looks good to me, just one minor suggestion.

**Comment by RByers** [comment]
LGTM, but needs to be merged with master to resolve the conflict.
